src/main/java/com/mineinabyss/bonfire/ecs/systems/BonfireEffectSystem.kt

```diff
@@ -0,0 +1,36 @@
+package com.mineinabyss.bonfire.ecs.systems
+
+import com.mineinabyss.bonfire.config.BonfireConfig
+import com.mineinabyss.bonfire.ecs.components.BonfireEffectArea
+import com.mineinabyss.bonfire.extensions.isBonfireModel
+import com.mineinabyss.geary.ecs.api.systems.TickingSystem
+import com.mineinabyss.geary.ecs.engine.iteration.QueryResult
+import org.bukkit.Particle
+import org.bukkit.entity.ArmorStand
+import org.bukkit.entity.Player
+
+
+class BonfireEffectSystem : TickingSystem(1) {
+    private val QueryResult.player by get<Player>()
+    private val QueryResult.effect by get<BonfireEffectArea>()
+
+    override fun QueryResult.tick() {
+        // Check if still near a bonfire
+        player.location.getNearbyLivingEntities(BonfireConfig.data.effectRadius).firstOrNull {
+            it is ArmorStand && it.isBonfireModel() && it.uniqueId == effect.uuid
+        }?.let {
+            player.location.world.spawnParticle(
+                listOf(Particle.SOUL, Particle.SOUL_FIRE_FLAME).random(),
+                player.location,
+                1,
```

Comment:
I generally like explicitly adding the parameter names when they go on newlines (you just have so much extra space).